cmd/snap-bootstrap/cmd_initramfs_mounts.go

```diff
@@ -1228,6 +1241,18 @@ func mountNonDataPartitionMatchingKernelDisk(dir, fallbacklabel string) error {
 		partSrc = filepath.Join("/dev/disk/by-label", fallbacklabel)
 	}
 
+	// The partition uuid is read from the EFI variables. At this point
+	// the kernel may not have initialized the storage HW yet so poll
+	// here.
+	if !osutil.FileExists(filepath.Join(dirs.GlobalRootDir, partSrc)) {
+		pollWait := 50 * time.Millisecond
+		pollIterations := 1200
+		logger.Noticef("waiting up to %v for %v to appear", time.Duration(pollIterations*int(pollWait)), partSrc)
```

Comment:
not sure we need to specify the max time that we will wait in two places, this message with the other one in the error case will look like this in the log:
```
waiting for up to 1 minute for /dev/foo to appear
no /dev/foo after waiting for 1 minute
```

I think it would read better to only include the max timeout in the error case i.e.

```
waiting for /dev/foo to appear
no /dev/foo after waiting for 1 minute
```

and then in the happy case it just reads

```
waiting for /dev/foo to appear
<proceeds to the rest of things>
```

Reply:
this is fine for a followup BTW